mula/scheduler/models/tasks.py

```diff
@@ -57,10 +57,39 @@ class Task(BaseModel):
 
     status: TaskStatus
 
-    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
+    meta: dict | None = None
+
+    # Status transition timestamps
+    pending_at: datetime | None = Field(default_factory=lambda: datetime.now(timezone.utc))
+    queued_at: datetime | None = None
+    dispatched_at: datetime | None = None
+    running_at: datetime | None = None
+    completed_at: datetime | None = None
 
+    created_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
     modified_at: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
```

Comment:
Is this needed if the only state changes are for the status that you use the ..._at fields for?